pkg/nodeagent/controller/operatingsystemconfig/changes.go

```diff
@@ -163,6 +222,119 @@ func computeOperatingSystemConfigChanges(log logr.Logger, fs afero.Afero, newOSC
+	}
+
+	return kubeletConfigCodec.Decode(kubeletConfigFile.Content.Inline)
+}
+
+// ComputeKubeletConfigChange computes changes in the kubelet configuration relevant for in-place updates.
+// This function needs to be updated when the kubelet configuration triggers in https://github.com/gardener/gardener/blob/master/docs/usage/shoot-operations/shoot_updates.md#rolling-update-triggers are changed.
+func ComputeKubeletConfigChange(oldConfig, newConfig *kubeletconfigv1beta1.KubeletConfiguration) (bool, bool, error) {
+	var (
+		cpuManagerPolicyChanged = oldConfig.CPUManagerPolicy != newConfig.CPUManagerPolicy
+		oldRelevantEvictionHard = make(map[string]string)
+		newRelevantEvictionHard = make(map[string]string)
+	)
+
+	// Copy only relevant config values for comparison.
+	if oldConfig.EvictionHard != nil {
+		oldRelevantEvictionHard[components.MemoryAvailable] = oldConfig.EvictionHard[components.MemoryAvailable]
+		oldRelevantEvictionHard[components.ImageFSAvailable] = oldConfig.EvictionHard[components.ImageFSAvailable]
+		oldRelevantEvictionHard[components.ImageFSInodesFree] = oldConfig.EvictionHard[components.ImageFSInodesFree]
+		oldRelevantEvictionHard[components.NodeFSAvailable] = oldConfig.EvictionHard[components.NodeFSAvailable]
+		oldRelevantEvictionHard[components.NodeFSInodesFree] = oldConfig.EvictionHard[components.NodeFSInodesFree]
+	}
+
+	if newConfig.EvictionHard != nil {
+		newRelevantEvictionHard[components.MemoryAvailable] = newConfig.EvictionHard[components.MemoryAvailable]
+		newRelevantEvictionHard[components.ImageFSAvailable] = newConfig.EvictionHard[components.ImageFSAvailable]
+		newRelevantEvictionHard[components.ImageFSInodesFree] = newConfig.EvictionHard[components.ImageFSInodesFree]
+		newRelevantEvictionHard[components.NodeFSAvailable] = newConfig.EvictionHard[components.NodeFSAvailable]
+		newRelevantEvictionHard[components.NodeFSInodesFree] = newConfig.EvictionHard[components.NodeFSInodesFree]
+	}
+
+	if !maps.Equal(oldRelevantEvictionHard, newRelevantEvictionHard) {
+		return true, cpuManagerPolicyChanged, nil
+	}
+
+	oldReserved, err := sumResourceReservations(oldConfig.KubeReserved, oldConfig.SystemReserved)
+	if err != nil {
+		return false, cpuManagerPolicyChanged, fmt.Errorf("failed to sum resource reservations for old kubelet config: %w", err)
+	}
+	newReserved, err := sumResourceReservations(newConfig.KubeReserved, newConfig.SystemReserved)
+	if err != nil {
+		return false, cpuManagerPolicyChanged, fmt.Errorf("failed to sum resource reservations for new kubelet config: %w", err)
+	}
+
+	return !maps.Equal(oldReserved, newReserved), cpuManagerPolicyChanged, nil
+}
+
+func sumResourceReservations(left, right map[string]string) (map[string]string, error) {
+	if left == nil {
+		return right, nil
+	} else if right == nil {
+		return left, nil
+	}
+
+	out := make(map[string]string)
+
+	for _, resource := range []string{"cpu", "memory", "ephemeral-storage", "pid"} {
```

Comment:
Do we still need to hard-code these here? Can't we take all that exist into account?

Reply:
I am considering only the fields considered in the original hash calculation.
https://github.com/gardener/gardener/blob/da2950e223b4d2f51a32dd0c64859040499f4fe0/pkg/utils/gardener/shoot.go#L792-L794

Reply:
Okay, makes sense. Thankyou!